src/averageAges.js

```diff
@@ -37,7 +47,17 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const womenArray = people.filter(
+    (person, index, arr) => {
+      return person.sex === 'f'
+        && (!withChildren || arr.some(item => item.mother === person.name));
+    });
+
+  const years = womenArray.map(person => person.died - person.born);
+
+  const sumOfAge = years.reduce((sum, age) => sum + age, 0);
+
+  return sumOfAge / years.length;
```

Comment:
create a separate helper function to calculate average age
you do the same several times